src/app/frontend/deploy/deploy_controller.js

```diff
@@ -43,10 +41,19 @@ export default class DeployController {
     this.namespaces = namespaces.namespaces;
 
     /**
-     * Currently chosen namespace.
-     * @export {(string|undefined)}
+     * Contains the selected directive's controller which has its own deploy logic
+     *
+     * Initialized from the template.
+     * @export {{deploy:function()}|undefined}
+     */
+    this.detail;
```

Comment:
I can modify [upload_controller](https://github.com/kubernetes/dashboard/blob/master/src/app/frontend/deploy/upload_controller.js#L30) in the same manner. :)